packages/launch/src/focused-launch/index.tsx

```diff
@@ -57,22 +41,27 @@ const FocusedLaunch: React.FunctionComponent = () => {
 	}, [ hasSelectedDomain, domainSuggestionFromCart, setDomain ] );
 
 	// @TODO: extract to some hook for reusability (Eg: use-products-from-cart)
-	// If there is no selected plan, but there is a plan in cart,
+	// If there is no selected plan and the site has no paid plan, but there is a plan in cart,
 	// set the plan from cart as the selected plan.
 	const planProductIdFromCart = usePlanProductIdFromCart();
 	const { setPlanProductId } = useDispatch( LAUNCH_STORE );
 
 	React.useEffect( () => {
-		if ( ! selectedPlanProductId && planProductIdFromCart ) {
+		if ( ! selectedPlanProductId && planProductIdFromCart && ! hasPaidPlan ) {
 			setPlanProductId( planProductIdFromCart );
 		}
-	}, [ selectedPlanProductId, planProductIdFromCart, setPlanProductId ] );
+	}, [ selectedPlanProductId, planProductIdFromCart, setPlanProductId, hasPaidPlan ] );
+
+	// If there is a purchased plan, remove any selected plan from Launch store
+	const { unsetPlanProductId } = useDispatch( LAUNCH_STORE );
+	React.useEffect( () => {
+		if ( hasPaidPlan ) {
+			unsetPlanProductId();
+		}
+	}, [ hasPaidPlan, unsetPlanProductId ] );
```

Comment:
Context: `unsetPlanProductId()` is called in situations where user may have previously used the launch flow to pick a paid plan, but then goes back customizer to actually purchase a paid plan. At this stage, the site is has a paid plan but it is not launched yet. So when launch modal reopens, we need to unset the selected plan product id. @razvanpapadopol Can you verify if this is one of the situations?

Reply:
Yes, exactly that's the case here. It's also one of the cases mentioned in https://github.com/Automattic/wp-calypso/issues/49958